core/assertion.ts

```diff
@@ -187,7 +187,9 @@ export class AssertionContext implements ICommonContext {
   }
 
   public name(): string {
-    return this.assertion.proto.target.name;
+    return this.assertion.session.finalizeName(
+      this.assertion.proto.target.name
+    );
```

Comment:
this is confusing - i am guessing you branched off your other branch? i am going to hold off reviewing this until that's in

(or rebase this branch on top of `main` without the unrelated commits)

Reply:
Yes, have merged that in and rebased this one now.